NGINX/nginx/ingest/parser.yml

```diff
@@ -0,0 +1,180 @@
+      properties:
+        pattern: '%{ERROR}|%{COMBINED}'
+        custom_patterns:
+          NGINXURL: '(%{URIPROTO:url_scheme}://)?(?:%{USER:url_username}(?::[^@]*)?@)?(?:%{URIHOST:url_domain})?(?:%{URIPATHPARAM:url_path})'
+          ERROR: '%{YEAR}/%{MONTHNUM}/%{MONTHDAY} %{HOUR}:%{MINUTE}:%{SECOND} \[%{WORD}\] %{NUMBER}#%{NUMBER}: (\*%{NUMBER} )?(%{DATA}client:\s+%{IP:source_address})?(%{DATA}server:\s+%{IPORHOST:destination_address})?(%{DATA}request: \"%{WORD:request_method} %{NGINXURL:url_original} HTTP/%{NUMBER:http_version}\")?(%{DATA}upstream: \"%{URIPROTO}://(?:%{USER}(?::[^@]*)?@)?(?:%{IPORHOST:url_name}(?::%{POSINT:url_port})?)?(?:%{URIPATHPARAM})?\")?%{GREEDYDATA}'
+          COMBINED: '(%{IPORHOST:destination_address} )?%{IP:source_address} - (-|%{DATA:source_user_name})? \[%{HTTPDATE:time_local}\] \"%{WORD:request_method} %{NOTSPACE:url_original} HTTP/%{NUMBER:http_version}\" %{NUMBER:response_status} (-|%{NUMBER:body_bytes_sent})? \"(-|%{DATA:http_referrer})?\" \"%{DATA:http_user_agent}\"( \"%{DATA}\")?( \"%{DATA}\")?( \"%{DATA}\")?( \"%{DATA:http_response_mimetype}\")?'
+    filter: '{{original.message | re_match("\{(?:[^{}]|())*\}") == False}}'
+  - name: parsed_event
+    external:
+      name: json.parse-json
+    filter: '{{original.message | re_match("\{(?:[^{}]|())*\}")}}'
+  - name: set_ecs_fields
+  - name: set_fields
+
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          event.kind: 'event'
+          event.category: ['web']
+          event.dataset: 'access'
+          observer.product: 'nginx'
+          observer.type: 'WEB server'
+          observer.vendor: 'F5'
+
+  set_fields:
+    actions:
+      - set:
+          '@timestamp': >
+            {%- set elt = parsed_event.message.time_local.split('/') -%}
+            {%- set month = elt[1] -%}
+            {%- set months = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sept','Oct','Nov','Dec'] -%}
+            {%- set monthnum = months.index(month)+1 -%}
+            {%- set rest = elt[2].split(':') -%}
+            {%- set year = rest[0] -%}
+            {%- set sec = rest[3].split(' ')[0] -%}
+            {%- set Z = rest[3].split(' ')[1][1:3] | int -%}
+            {%- set z = rest[3].split(' ')[1][3:] |int -%}
+            {%- set hour = rest[1] | int -%}
+            {%- set hour = hour-Z -%}
+            {%- if hour < 10 -%}
+              {%- set hour = hour | string() -%}
+              {%- set hour = '0' + hour -%}
+            {%- endif -%}
+            {%- set min = rest[2] | int -%}
+            {%- set min = min-z -%}
+            {%- if min < 10 -%}
+              {%- set min = min | string() -%}
+              {%- set min = '0' + min -%}
+            {%- endif -%}
+            {%- if monthnum < 10 -%}
+              {{year}}-0{{monthnum}}-{{elt[0]}}T{{hour}}:{{min}}:{{sec}}.000000Z
+            {%- else -%}
+              {{year}}-{{monthnum}}-{{elt[0]}}T{{hour}}:{{min}}:{{sec}}.000000Z
+            {%- endif -%}
```

Comment:
Why not use the stage `date.parse` with format '%d/%b/%Y:%H:%M:%S %z' to parse the time_local?